hashintel/hash
# Add updated BP getAggregateEntities hook

## 🌟 What is the purpose of this PR?

This introduces a new getAggregateEntities BP hook, returning a `Subgraph` object.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1201095311341924/1203226609245274/f) _(internal)_

## 🚫 Blocked by




- [x] #1242 

## 🔍 What does this change?

- Adds useBlockProtocolAggregateEntities hook
- Removes the old hook
- Adds an example usage of `aggregateEntities` in the `entity-editor` demo page

## 📜 Does this require a change to the docs?

When we come to discuss the changes to the BP hook signatures we will need to. In the meantime I don't believe this is captured anywhere.

## ⚠️ Known issues

Links are broken and should probably not be used from the subgraph.

## 🐾 Next steps

- Create helper methods to get all Users and all Orgs

## ❓ How to test this?

1.  Checkout the branch and follow README to setup dev environment
1.  Check the `/entity-editor` route works there's a section in it called "Aggregate Entities" which prints the results of the hook
1.  Check the actual entity editor and type editor routes and make sure pages and such still work

## 📹 Demo

<img width="1132" alt="image" src="https://user-images.githubusercontent.com/25749103/197838114-5a08ca10-5a93-4ab5-b5d9-aa4a4bc97a46.png">
